compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ASTBuilderUtil.java

```diff
@@ -852,6 +854,16 @@ public static BInvokableSymbol duplicateInvokableSymbol(BInvokableSymbol invokab
         dupFuncSymbol.type = dupInvokableType;
         dupFuncSymbol.dependentGlobalVars = invokableSymbol.dependentGlobalVars;
 
+        if (!Symbols.isFlagOn(invokableSymbol.flags, Flags.ATTACHED)) {
+            return dupFuncSymbol;
+        }
+        List<BAttachedFunction> attachedFuncs = ((BStructureTypeSymbol) invokableSymbol.owner).attachedFuncs;
```

Comment:
I guess this modifies the BIR. maybe we should add a BIR test case to verify this.